test/e2e/baremetal_test.go

```diff
@@ -4,54 +4,56 @@ import (
 	"fmt"
 	"testing"
 
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	configv1 "github.com/openshift/api/config/v1"
+	configapiv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/cluster-image-registry-operator/test/framework"
 )
 
-const (
-	installerConfigNamespace = "kube-system"
-	installerConfigName      = "cluster-config-v1"
-)
-
 func TestBaremetalDefaults(t *testing.T) {
 	client := framework.MustNewClientset(t, nil)
 
-	// We don't have CI for baremetal, so let's fake this environment.
-	clusterConfig, err := client.ConfigMaps(installerConfigNamespace).Get(installerConfigName, metav1.GetOptions{})
+	infrastructureConfig, err := client.Infrastructures().Get("cluster", metav1.GetOptions{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	err = client.ConfigMaps(installerConfigNamespace).Delete(installerConfigName, nil)
-	if err != nil {
-		t.Fatal(err)
+
+	if infrastructureConfig.Status.PlatformStatus.Type != configapiv1.BareMetalPlatformType {
+		t.Skip("skipping on non-BareMetal platform")
 	}
-	defer func() {
-		err = client.ConfigMaps(installerConfigNamespace).Delete(installerConfigName, nil)
-		if err != nil && !errors.IsNotFound(err) {
-			panic(fmt.Errorf("unable to delete fake cluster config: %s", err))
-		}
-		clusterConfig.ResourceVersion = "" // should not be set on objects to be created
-		_, err = client.ConfigMaps(installerConfigNamespace).Create(clusterConfig)
-		if err != nil {
-			panic(fmt.Errorf("unable to restore cluster config: %s; %s=%v", err, installerConfigName, clusterConfig))
-		}
-	}()
-	_, err = client.ConfigMaps(installerConfigNamespace).Create(&corev1.ConfigMap{
+
+	_, err = client.Infrastructures().Update(&configapiv1.Infrastructure{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: installerConfigName,
+			Name:            infrastructureConfig.Name,
+			ResourceVersion: infrastructureConfig.ResourceVersion,
 		},
-		Data: map[string]string{
-			"install-config": `apiVersion: v1beta4`,
+		Status: configapiv1.InfrastructureStatus{
+			InfrastructureName: infrastructureConfig.Status.InfrastructureName,
+			PlatformStatus: &configapiv1.PlatformStatus{
+				Type: configapiv1.BareMetalPlatformType,
+			},
+			EtcdDiscoveryDomain:  infrastructureConfig.Status.EtcdDiscoveryDomain,
+			APIServerURL:         infrastructureConfig.Status.APIServerURL,
+			APIServerInternalURL: infrastructureConfig.Status.APIServerInternalURL,
```

Comment:
What purpose does this serve? Doesn't this information already exist for a baremetal cluster?

Reply:
Originally, this test was deleting the installConfig to replicate a baremetal install, but I was updating it to use infrastructures, which you can't delete.  And if you set the infrastructure ti baremetal, it seems to change a lot of stuff with the cluster and makes the test fail.  So this test is currently being skipped until we have an actual baremetal install to test on.

Reply:
+1 to skipping unless we are on baremetal - need to check if we have a baremetal e2e suite (we should)

My inclination is to follow what happens with AWS - we don't mess with the infrastructure config, just the options that are available for baremetal.

Reply:
@adambkaplan not sure I am following your train of thought here.  Can you elaborate?

Reply:
@dmage can weigh in - my understanding is that the current baremetal tests run on AWS, but we fake things out so the registry operator thinks it is on baremetal.

I confirmed with @stevekuznetsov that we now have CI cluster profiles for baremetal and vSphere, so we should update our tests so that they run on the appropriate platform, and only verify settings that are specific to the storage provider for the given platform. Separately, we should add new e2e suites (`e2e-baremetal`, `e2e-baremetal-operator`, etc.)

Reply:
Got it.  Updated the baremetal test to only run on baremetal, and assume that it is running on baremetal.